homeassistant/components/osc/__init__.py

```diff
@@ -0,0 +1,118 @@
+"""The OSC (Open Sound Control) integration."""
+import logging
+
+from pythonosc.udp_client import SimpleUDPClient
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant, ServiceCall
+from homeassistant.helpers import config_validation as cv
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+CONFIG_SCHEMA = cv.config_entry_only_config_schema(DOMAIN)
+
+PLATFORMS: list[Platform] = [Platform.SENSOR]
+
+# Define the service schema
+SERVICE_OSC_SEND_SCHEMA = vol.Schema(
+    {
+        vol.Required("entity_id"): cv.string,
+        vol.Required("value"): cv.string,
+        vol.Required("data_type"): vol.In(["float", "int"]),
+    }
+)
+
+
+def setup(hass, config):
+    """Set up is called when Home Assistant is loading our component."""
+
+    hass.data.setdefault(DOMAIN, {})
+
+    def handle_send_osc(call: ServiceCall):
```

Comment:
another option for this service is to create an entity service which will call a function on the entity. It will simplify some things for you. Search for `entity_service` in the code, I can't remember the function names offhand but you should see examples of this being done in other integrations